commercetools-internal-processors/src/main/java/io/sphere/sdk/annotations/processors/generators/BaseAbstractGenerator.java

```diff
@@ -98,32 +98,27 @@ protected String escapeJavaKeyword(final String name) {
                 .collect(Collectors.toList());
     }
 
-    protected MethodSpec createGetMethod(final ExecutableElement propertyMethod) {
-        final MethodSpec.Builder builder = createGetMethodBuilder(propertyMethod);
-        return builder.build();
-    }
-
-    protected MethodSpec.Builder createGetMethodBuilder(final ExecutableElement propertyMethod) {
-        final MethodSpec.Builder builder = MethodSpec.methodBuilder(propertyMethod.getSimpleName().toString())
+    protected MethodSpec createGetMethod(final PropertyGenModel property) {
+        final MethodSpec.Builder builder = MethodSpec.methodBuilder(property.getMethodName())
                 .addModifiers(Modifier.PUBLIC)
-                .returns(TypeName.get(propertyMethod.getReturnType()))
-                .addCode("return $L;\n", escapeJavaKeyword(PropertyGenModel.getPropertyName(propertyMethod)));
-        copyNullableAnnotation(propertyMethod, builder);
-        return builder;
+                .returns(property.getType())
+                .addCode("return $L;\n", property.getJavaIdentifier());
+        copyNullableAnnotation(property, builder);
+        copyDeprecatedAnnotation(property, builder);
+        copyJsonAnnotation(property, builder);
+        return builder.build();
     }
 
-    protected void copyJsonAnnotation(final ExecutableElement propertyMethod, final MethodSpec.Builder builder) {
-        final JsonProperty jsonProperty = propertyMethod.getAnnotation(JsonProperty.class);
-        final String jsonName = jsonProperty != null ? jsonProperty.value() : null;
```

Comment:
😄  i like the code amelioration here, it's not obvious to notice to notice the duplication here